twindb_backup/destination/ssh.py

```diff
@@ -124,64 +113,93 @@ def get_stream(self, path):
-        cmd = self._ssh_command + [
+        cmd = [
             "echo {raw_status} > "
             "{status_file}".format(raw_status=raw_status,
                                    status_file=self.status_path)
         ]
-        proc = Popen(cmd)
-        _, cerr = proc.communicate()
-        if proc.returncode:
+        stdout, _ = self._execute_commnand(cmd)
+        if stdout.channel.recv_exit_status():
             LOG.error('Failed to write backup status')
-            LOG.error(cerr)
             exit(1)
         return status
 
     def _read_status(self):
 
         if self._status_exists():
-            cmd = self._ssh_command + ["cat %s" % self.status_path]
-            proc = Popen(cmd, stdout=PIPE, stderr=PIPE)
-            cout, cerr = proc.communicate()
-            if proc.returncode:
-                LOG.error('Failed to read backup status: %d: %s',
-                          proc.returncode,
-                          cerr)
+            cmd = ["cat %s" % self.status_path]
+            stdout, _ = self._execute_commnand(cmd)
+            if stdout.channel.recv_exit_status():
+                LOG.error('Failed to read backup status: %d', stdout.channel.recv_exit_status())
                 exit(1)
-            return json.loads(base64.b64decode(cout))
+            return json.loads(base64.b64decode(stdout.read()))
         else:
             return self._empty_status
 
     def _status_exists(self):
-        cmd = self._ssh_command + \
-              ["bash -c 'if test -s %s; "
+        cmd = ["bash -c 'if test -s %s; "
                "then echo exists; "
                "else echo not_exists; "
                "fi'" % self.status_path]
 
         try:
             LOG.debug('Running %r', cmd)
-            proc = Popen(cmd, stdout=PIPE, stderr=PIPE)
-            cout, cerr = proc.communicate()
-            if proc.returncode:
-                LOG.error('Failed to read backup status: %d: %s',
-                          proc.returncode,
-                          cerr)
+            stdout, _ = self._execute_commnand(cmd)
+            output = stdout.read()
+            if stdout.channel.recv_exit_status():
+                LOG.error('Failed to read backup status: %d',
+                          stdout.channel.recv_exit_status())
                 exit(1)
-            if cout.strip() == 'exists':
+            if output.strip() == 'exists':
                 return True
-            elif cout.strip() == 'not_exists':
+            elif output.strip() == 'not_exists':
                 return False
             else:
-                raise DestinationError('Unrecognized response: %s' % cout)
+                raise DestinationError('Unrecognized response: %s' % output)
 
         except OSError as err:
             LOG.error('Failed to run %s: %s', " ".join(cmd), err)
             exit(1)
+
     def share(self, url):
         super(Ssh, self).share(url)
+
+    def _execute_commnand(self, cmd):
+        """Execute ssh command"""
```

Comment:
Could you add docstring with params, their types, return value, its type and exceptions (if any)?